lib/improver/tests/nbhood/circular_kernel/test_GeneratePercentilesFromACircularNeighbourhood.py

```diff
@@ -0,0 +1,656 @@
+        expected = np.array(
+            [[[[[1., 1., 1., 1., 1.],
+                [1., 1., 0.2, 1., 1.],
+                [1., 0.2, 0.2, 0.2, 1.],
+                [1., 1., 0.2, 1., 1.],
+                [1., 1., 1., 1., 1.]]],
+              [[[1., 1., 1., 1., 1.],
+                [1., 1., 0.4, 1., 1.],
+                [1., 0.4, 0.4, 0.4, 1.],
+                [1., 1., 0.4, 1., 1.],
+                [1., 1., 1., 1., 1.]]],
+              [[[1., 1., 1., 1., 1.],
+                [1., 1., 0.8, 1., 1.],
+                [1., 0.8, 0.8, 0.8, 1.],
+                [1., 1., 0.8, 1., 1.],
+                [1., 1., 1., 1., 1.]]]]])
+        cube = set_up_cube(
+            zero_point_indices=((0, 0, 2, 2),), num_time_points=1,
+            num_grid_points=5)
+        percentiles = np.array([5, 10, 20])
+        radius = 2000.
+        result = (
+            GeneratePercentilesFromACircularNeighbourhood(
+                percentiles=percentiles).run(
+                    cube, radius))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_single_point_range_0(self):
+        """Test behaviour with zero range."""
+        cube = self.cube
+        radius = 0.
+        msg = "Distance of {0}m gives zero cell extent".format(radius)
+        with self.assertRaisesRegexp(ValueError, msg):
+            GeneratePercentilesFromACircularNeighbourhood().run(cube, radius)
+
+    def test_point_pair(self):
+        """Test behaviour for two nearby non-zero grid cells."""
+        expected = np.array(
+            [[[[[1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 0., 0., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.]]],
+              [[[1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.]]],
+              [[[1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.]]]]])
+        cube = set_up_cube(
+            zero_point_indices=((0, 0, 2, 2), (0, 0, 2, 1)), num_grid_points=5)
+        percentiles = np.array([25, 50, 75])
+        radius = 2000.
+        result = (
+            GeneratePercentilesFromACircularNeighbourhood(
+                percentiles=percentiles).run(
+                    cube, radius))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_number_of_percentiles_equals_number_of_points(self):
+        """Test when the number of percentiles is equal to the number of points
+        used to construct the percentiles."""
+        expected = np.array(
+            [[[[[1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 0.2, 1., 1., 1.],
+                [1., 1., 0.2, 0.2, 0.2, 1., 1.],
+                [1., 1., 1., 0.2, 1.,  1., 1.],
```

Comment:
Pylint complains you have two spaces after the 4th "1." in this line.

Reply:
Changed.